standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/jdbc/MultiDataSourceJdbcResource.java

```diff
@@ -46,19 +57,23 @@ public class MultiDataSourceJdbcResource {
 
   private static final Logger LOG = LoggerFactory.getLogger(MultiDataSourceJdbcResource.class);
 
-  private final ThreadLocal<ContextNode<String>> threadLocal = new ThreadLocal<>();
+  private final StackThreadLocal<String> threadLocal = new StackThreadLocal<>();
```

Comment:
should we inline StackThreadLocal impl here? it's not used anywhere else, isn't it?